### PR TITLE
[develop]: Add plotting and template variable documentation

### LINKED ISSUE #525: Add User's Guide documentation on how to use the plotting scripts in the workflow
## Description
Currently, the ReadTheDocs (RTD) documentation does not reflect the addition of the plot_allvars task to the workflow in PR #482. 

## Solution
Update the following files to reflect the changes made in PR #482 : 
- RunSRW.rst --> Update the plotting section to reflect that plotting scripts are now part of the workflow. Migrate relevant information from the Graphics chapter to the RunSRW chapter. 
- Delete the Graphics.rst file once relevant info is incorporated into RunSRW.rst. 
- ConfigWorkflow.rst --> Add variables and descriptions from the config_defaults.yaml documentation to this file.

### PR DESCRIPTION
**Pull request by gspetro-NOAA**
## DESCRIPTION OF CHANGES: 
- Added a plotting configuration section to RunSRW.rst
- Deleted Graphics.rst (all relevant info has been added to RunSRW.rst) and removed/updated references to the Graphics chapter throughout the docs. 
- Updated ConfigWorkflow.rst with variables related to the addition of plotting to the workflow
- Added chapter on Template Variables based on @gsketefian 's closed Issue #198. (Includes updates due to the switch to a python workflow and a concrete example (COMOUT_REF) from the plotting tasks.)

### Type of change
- [X] New feature (non-breaking change which adds functionality)
- [X] This change requires a documentation update

## TESTS CONDUCTED: 
None required (all documentation). I built the docs on my RTD account: https://srw-ug.readthedocs.io/en/text-plotting/index.html

*Note that ReadTheDocs did dependencies updates on 1/26/22 and 1/4/23 that seem to be causing issues rendering bullet points properly. This will make any bulleted lists in our docs ugly, but the information is still there. I don't think the issue relates to anything I did, since the bullets are not rendering properly anywhere in the docs (including lists I didn't touch). There was also no problem with my RTD builds from 12/22/22.*

- [X] hera.intel (out of the box case)

## DEPENDENCIES:
N/A.

## DOCUMENTATION:
All documentation.

## ISSUE: 
Issue #525 
Issue #482 
Issue #230 

## CHECKLIST
- [X] My code follows the style guidelines in the Contributor's Guide
- [X] I have performed a self-review of my own code using the Code Reviewer's Guide
- [ ] I have commented my code, particularly in hard-to-understand areas N/A
- [X] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [ ] My changes do not require updates to the documentation (explain). N/A
- [X] My changes generate no new warnings
- [ ] New and existing tests pass with my changes N/A
- [X] Any dependent changes have been merged and published N/A

## CONTRIBUTORS (optional): 
@danielabdi-noaa @gsketefian @EdwardSnyder-NOAA

### CODE REVIEW

File: docs/UsersGuide/source/ConfigWorkflow.rst [hunk 1]
```diff
@@ -1134,6 +1140,9 @@ These parameters are associated with the fixed (i.e., static) files. On `Level 1
 ``FIXlut``: (Default: "")
    System directory where the lookup tables for optics properties are located.
 
+``FIXshp``: (Default: "")
+   System directory where the graphics shapefiles are located. On Level 1 systems, these are set within the machine files. Users on other systems will need to provide the path to the directory that contains the *Natural Earth* shapfiles.
```
Comment (MichaelLueken) [inline docs/UsersGuide/source/ConfigWorkflow.rst]: Should *Natural Earth* shapfiles be shapefiles?

File: docs/UsersGuide/source/ConfigWorkflow.rst [hunk 2]
```diff
@@ -1726,6 +1735,49 @@ Non-default parameters for the ``run_enspointvx_prob`` task are set in the ``tas
...
+``WTIME_PLOT_ALLVARS``: (Default: 01:00:00)
+   Maximum time for the task to complete.
+
+``MAXTRIES_PLOT_ALLVARS``: (Default: 1)
+   Maximum number of times to attempt the task.
+
+Additional Parameters
+------------------------
+
+Typially, the following parameters must be set explicitly by the user in the configuration file (``config.yaml``) when executing the plotting tasks.
```
Comment (MichaelLueken) [inline docs/UsersGuide/source/ConfigWorkflow.rst]: Should Typially be Typically?

**Review by MichaelLueken — COMMENTED** [review]
@gspetro-NOAA Thank you very much for updating the plotting and template variable documentation!  For the most part, everything looks great!  I have noted a few minor spelling issues in `ConfigWorkflow.rst`.

File: docs/UsersGuide/source/ConfigWorkflow.rst (hunk 1 above)
Comment (gspetro-NOAA) [reply docs/UsersGuide/source/ConfigWorkflow.rst]: Good catch! Apparently my brain was on vacation leading up to Christmas. ;)

File: docs/UsersGuide/source/ConfigWorkflow.rst (hunk 2 above)
Comment (gspetro-NOAA) [reply docs/UsersGuide/source/ConfigWorkflow.rst]: Yes! (See above ;) )

File: docs/UsersGuide/source/RunSRW.rst [hunk 3]
```diff
@@ -534,6 +535,77 @@ Valid values for configuration variables should be consistent with those in the
...
+
+.. _Cartopy:
+
+Cartopy Shapefiles
+^^^^^^^^^^^^^^^^^^^^^
+
+The Python plotting tasks require a path to the directory where the Cartopy Natural Earth shapefiles are located. The medium scale (1:50m) cultural and physical shapefiles are used to create coastlines and other geopolitical borders on the map. On `Level 1 <https://github.com/ufs-community/ufs-srweather-app/wiki/Supported-Platforms-and-Compilers>`__ systems, this path is already set in the system's machine file using the variable ``FIXshp``. Users on other systems will need to download the shapefiles and update the path of ``$FIXshp`` in the machine file they are using (e.g., ``$HOME/ush/machine/macos.yaml`` for a generic MacOS system, where ``$HOME`` is the path to the ``ufs-srweather-app`` directory). The subset of shapefiles required for the plotting task can be obtained from the `SRW Data Bucket <https://noaa-ufs-srw-pds.s3.amazonaws.com/NaturalEarth/NaturalEarth.tgz>`__. The full set of medium-scale (1:50m) Cartopy shapefiles can be downloaded `here <https://www.naturalearthdata.com/downloads/>`__.
```
Comment (MichaelLueken) [inline docs/UsersGuide/source/RunSRW.rst]: @gspetro-NOAA Would it be best to replace `$HOME` with `$SRW` in this line?  The `$SRW` environment variable should already be set in the BuildSRW chapter, so it should already be created before the RunSRW chapter.

**Review by MichaelLueken — COMMENTED** [review]
@gspetro-NOAA I have one more comment, this time in `RunSRW.rst` - would it be best to replace `$HOME` with `$SRW` on line 566?  In the `BuildSRW` chapter, it suggests that users create an environment variable pointing to their cloned SRW App and call it `$SRW`.  It might be best to push for the use of this environment variable.

**Review by MichaelLueken — APPROVED** [review]
@gspetro-NOAA Thank you very much for addressing my concerns!  Approving this work now.

**Review by danielabdi-noaa — APPROVED** [review]
These documentation changes look good to me! Whatever is going on with the workflow for plotting is not related to the docs so approving!

**Pull request merged**